addons/account/models/account_journal.py

```diff
@@ -484,6 +484,28 @@ def _check_auto_post_draft_entries(self):
     def _onchange_type(self):
         self.refund_sequence = self.type in ('sale', 'purchase')
 
+    @api.onchange('payment_sequence')
+    def _onchange_payment_sequence(self):
+        if self.payment_sequence:
+            new_payment = self.env['account.payment'].new({
+                'journal_id': self.id,
+            })
+            new_payment_seq = new_payment.move_id._get_last_sequence() or ""
+            new_payment_seq_param = new_payment.move_id._get_sequence_format_param(new_payment_seq)
+            new_stmt = self.env['account.bank.statement.line'].new({
+                'journal_id': self.id,
+                'amount': 20,
+            })
+            new_stmt_seq = new_stmt.move_id._get_last_sequence()
+            if not new_stmt_seq and new_payment_seq_param[1]['prefix1'] == self.code + '/':
```

Comment:
Why don't you just compare the first element of `_get_sequence_format_param` called on the last sequence for the payment and the statement?
If they are equal, we have an issue. If not, everything is good.

You can't use `'/'`

Reply:
we only have an issue when we have 'no statement line (then, no last sequence for `new_stmt`) and payment move with default sequence'

Reply:
No. If you have
* Payment `BNK/2020/0001`
* Statement `BNK/2020/0002`

Then you change the parameter on the journal.
The next payment you make will try to get the number `BNK/2020/0002`, which will trigger the following constraint: 
```sql
                CREATE UNIQUE INDEX account_move_unique_name
                                 ON account_move(name, journal_id)
                              WHERE (state = 'posted' AND name != '/')
```

You could interchange the payment and the statement line in the example above.

If you don't have one of the 2 documents (yet), you might still suggest a conflicting name (but not necessarily).
The only way to know for sure what the next number will be, and if there is a chance of conflict one day is to use the same mechanism as the one setting the next sequence:
https://github.com/odoo/odoo/blob/4367935cbcd2daaba261bb87392798399ceb4079/addons/account/models/sequence_mixin.py#L247-L252

Reply:
If I have

- Payment `BNK/2020/0001`
- Statement `BNK/2020/0002`

Then I change the parameter on the journal.
The next payment I make will just get the number `BNK/2020/0003` ---> no issue

If I have

- Payment `BNK/2020/0001`

or even

- Payment `PBNK/2020/0001`
- Payment `BNK/2020/0002`
- Payment `BNK/2020/0001`

Then I change the parameter on the journal.
The next payment I make will try to get the number `BNK/2020/0001` ---> problem, that's why I wanted to check if there any payment move that still had the sequence prefix, checking the last or a newly created payment is not what I need.

I just pushed a test to show what's going on, because I think we misunderstand each other. For now, fix is not good anymore as I check only the last payment.